File: hw/chavez/u3/hw21AnimalFarm/javascript/model/Pig.js
import FarmAnimal from "./FarmAnimal.js";
import Cut from "./Cut.js";
import SlaughterHouse from "./SlaughterHouse.js";

export default class Pig extends FarmAnimal {

    cut() {
        console.log("Pig has been cut");

        const cuts = [
            new Cut("Loin"),
            new Cut("Belly"),
            new Cut("Feet")
        ];

        cuts.forEach(cut => console.log(cut.toString()));
    }

    sendToSlaughterHouse(name) {
        const slaughterHouse = new SlaughterHouse(name);
        slaughterHouse.slaughter(this);
    }
}

File: hw/chavez/u3/hw21AnimalFarm/python/model/Pig.py
from model.FarmAnimal import FarmAnimal
from model.Cut import Cut
from model.SlaughterHouse import SlaughterHouse

class Pig(FarmAnimal):

    def cut(self):
        print("Pig has been cut")

        cuts = [
            Cut("Loin"),
            Cut("Belly"),
            Cut("Feet")
        ]

        for cut in cuts:
            print(cut)

    def send_to_slaughter_house(self, name):
        slaughter_house = SlaughterHouse(name)
        slaughter_house.slaughter(self)